From 52dc7dce7a77b477a32f0735c01257a5e9309d86 Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Fri, 13 Jun 2014 18:54:38 -0400
Subject: [block] blk-mq: rename mq_flush_work struct request member

Message-id: <20140613185437.GC18176@redhat.com>
Patchwork-id: 83935
O-Subject: [RHEL7.1 PATCH v2 040/150] blk-mq: rename mq_flush_work struct request member
Bugzilla: 1105204
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Jeff Moyer <jmoyer@redhat.com>

BZ: 1105204

Breaks kABI but it will be fixed at the end of this patch series.

Upstream commit f88a164b72bd51fe4c89e06ac9939f2afe39c7ed
Author: Christoph Hellwig <hch@lst.de>
Date:   Wed Apr 16 09:44:58 2014 +0200

    blk-mq: rename mq_flush_work struct request member

    We will use this work_struct to requeue scsi commands from the
    completion handler as well, so give it a more generic name.

    Signed-off-by: Christoph Hellwig <hch@lst.de>
    Signed-off-by: Jens Axboe <axboe@fb.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/block/blk-flush.c b/block/blk-flush.c
index b1ca34e..bb73d70 100644
--- a/block/blk-flush.c
+++ b/block/blk-flush.c
@@ -134,7 +134,7 @@ static void mq_flush_run(struct work_struct *work)
 {
  struct request *rq;
 
- rq = container_of(work, struct request, mq_flush_work);
+ rq = container_of(work, struct request, requeue_work);
 
  memset(&rq->csd, 0, sizeof(rq->csd));
  blk_mq_insert_request(rq, false, true, false);
@@ -143,8 +143,8 @@ static void mq_flush_run(struct work_struct *work)
 static bool blk_flush_queue_rq(struct request *rq, bool add_front)
 {
  if (rq->q->mq_ops) {
-  INIT_WORK(&rq->mq_flush_work, mq_flush_run);
-  kblockd_schedule_work(&rq->mq_flush_work);
+  INIT_WORK(&rq->requeue_work, mq_flush_run);
+  kblockd_schedule_work(&rq->requeue_work);
   return false;
  } else {
   if (add_front)
diff --git a/include/linux/blkdev.h b/include/linux/blkdev.h
index 9e537d9..9931256 100644
--- a/include/linux/blkdev.h
+++ b/include/linux/blkdev.h
@@ -105,7 +105,7 @@ struct request {
 #endif
  union {
   struct call_single_data csd;
-  struct work_struct mq_flush_work;
+  struct work_struct requeue_work;
   unsigned long fifo_time;
  };
 
-- 
1.7.1